device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSessionHandler.java

```diff
@@ -153,7 +156,7 @@ public void onSessionLocalOpen(Event e)
     public void onSessionRemoteClose(Event e)
     {
         Session session = e.getSession();
-        if (session.getLocalState() == EndpointState.ACTIVE)
+        if (session.getLocalState() == EndpointState.ACTIVE || !isClosing)
```

Comment:
In my testing, there were times where the session was closing unexpectedly when the local state of the session wasn't ACTIVE yet, so I'm adding this additional flag to ensure we know when a session closure is expected or not. Our transport layer handles unexpected session closures differently than expected session closures, so it's important to get this distinction correct